weasyl/two_factor_auth.py

```diff
@@ -0,0 +1,349 @@
+"""
+Module for handling 2FA-related functions.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import re
+import urllib
+
+import bcrypt
+import pyotp
+from qrcodegen import QrCode
+
+from libweasyl import security
+from weasyl import define as d
+from weasyl import login
+
+# Number of recovery codes to provide the user
+_TFA_RECOVERY_CODES = 10
+# Work factor for bcrypt to be used in hashing each recovery code
+BCRYPT_WORK_FACTOR = 10
```

Comment:
This will block the time to save recovery codes, correct? i.e. the load time of the 'post' request to save codes and enable 2FA?

See comment about which recovery codes get saved. I just want to be careful here to make sure it's infeasible that a user saves a different set of codes than the ones they printed out.